services/webhook/tests/test_llm_client.py

```
@@ -400,13 +400,97 @@ def respond(url, **kwargs):
+
+def test_decide_deep_escalation_sample_is_deterministic() -> None:
+    ctx = {"repo": "o/r", "pr_number": 7, "head_sha": "abc123"}
+    a = lc.decide_deep_escalation(
+        [_hunk()], pr_context=ctx, sample_rate=1.0,
+        diff_line_threshold=99999, path_markers=(),
+    )
+    b = lc.decide_deep_escalation(
+        [_hunk()], pr_context=ctx, sample_rate=1.0,
+        diff_line_threshold=99999, path_markers=(),
+    )
+    assert a.escalate is True and b.escalate is True
```

Ruff (0.15.21)

[warning] Assertion should be broken down into multiple parts

Break down assertion into multiple parts

(PT018)